lib/addons/src/hooks/useCallbackRef.ts

```diff
@@ -0,0 +1,14 @@
+import { useState, useCallback, MutableRefObject } from 'react';
+
+// @TODO should be React.MutableRefObject....
+export function useCallbackRef<T extends HTMLElement = HTMLElement>() {
+  const [callbackRef, setRef] = useState<MutableRefObject<T | null>>(null);
+  const setCallbackRef = useCallback(
+    (ref: MutableRefObject<T | null>) => {
+      setRef(ref);
+    },
+    [setRef]
+  );
+
+  return { callbackRef, setCallbackRef };
+}
```

Comment:
I think this hook is pointless

Reply:
It is required for more complex use cases when it comes to DOM element positions

Every hook I have been looking at does it as well (not that bandwagon argumentation is a thing) - but tons of posts about it as well

I do not claim to be able to fully convince you with argumentation as I lack that kind of detailed in-depth knowledge though.

Can we have a chat about it and go in-depth @ndelangen?

Reply:
Hook is gone